### DISCUSSION TITLE
How to import css and js in _app.js

### DISCUSSION THREAD

**Discussion by arthurtalkgoal** [keystonejs/keystone#7513] [Feature requests]
For latest Next.js, to include global css and js, we need to import in _app.js.

However, the _app.js looks like reserved by keystone itself. How can we achieve it. 
E.g. the following will be an error:

```
// admin/pages/_app.js

import './styles/global.css';

// This default export is required in a new `pages/_app.js` file.
export default function MyApp({ Component, pageProps }) {
  return <Component {...pageProps} />
}
```

The error:

```

error - ./pages/_app.js
Error: error: Expected ';', got 'o'

  |
1 | export { default } from "../../../admin/pages/_app.js"o-not-use-will-break-in-patch/admin-ui/pages/App';
  |                                                       ^

Caused by:
    0: failed to process input file
    1: Syntax Error
error - Error: Cannot find module '<myfolder>/.keystone/admin/.next/server/pages-manifest.json'

```

**Comment by raveling** [comment]
@arthurtalkgoal currently this isn't possible. I'll change this discussion to a feature request for the future 🙂

**Comment by piofinn** [comment]
Chiming in to say that this would be a massive improvement. Currently we can specify custom Component Blocks in the Document editor, but without the ability to import styles we _have_ to use inline styles. If we could import our own styles, we could use e.g. design system components directly in the editor, giving content creators a much better idea of how the page will look when live.

**Reply by wkillerud** [reply]
It would be nice to have an official API for this, but for now I found a workaround:

- Write a post-install script to copy CSS and font files from `node_modules/` to `public/static/`.
- Hijack the [Custom Logo](https://keystonejs.com/docs/guides/custom-admin-ui-logo#custom-admin-ui-logo) feature to render a Logo, _and_ a bunch of `<link>` tags.

It's a bit _icky_, but seems to be working OK. Good enough for the proof-of-concept anyway 😄 

- Repo: https://github.com/wkillerud/keystone-starter-global-css
- Postinstall: https://github.com/wkillerud/keystone-starter-global-css/blob/main/postinstall.js
- Custom logo: https://github.com/wkillerud/keystone-starter-global-css/blob/main/admin/config.tsx

**Comment by VinSpee** [comment]
Ran into this today – thanks @wkillerud for the clever hack-around. IMO this is a large oversight.

**Comment by flexdinesh** [comment]
There's an open PR for this and progress is happening.

- #8020

**Reply by gautamsi** [reply]
There is still a caveat, the custom app does not actually replaces _app.js, it only imports custom app.js/tsx from within _app.js which still is complained by nextjs

**Comment by jhohlfeld** [comment]
I'm not sure if this fits in here but I ran over this thread while looking for a way to store static assets for my admin ui. 

Sure there is [local storage](https://keystonejs.com/docs/guides/images-and-files#storing-assets-in-local), but these assets are not accessible by next/image. My feeling is these static assets should go into the public folder of the admin NextJS app. However, this dir is being wiped on build 🤔.

Is there a good way to bring static assets into `.keystone/admin/public/`?

**Reply by Francoois** [reply]
You have probably find out since the time you asked, but you can use `getAdditionalFiles` from [ui.config](https://keystonejs.com/docs/config/config#ui) to move files to *.keystone*.
